cpp/src/arrow/acero/swiss_join.cc

```diff
@@ -2637,10 +2635,6 @@ class SwissJoin : public HashJoinImpl {
         static_cast<int64_t>(thread_id), key_batch, no_payload ? nullptr : &payload_batch,
         temp_stack)));
 
-    // Release input batch
-    //
-    input_batch.values.clear();
```

Comment:
This is just a stack variable so no need to explicitly release before returning.